gossip/infostore.go

```diff
@@ -146,26 +150,33 @@ func (is *infoStore) getInfo(key string) *info {
 // addInfo adds or updates an info in the infos map.
 //
 // Returns nil if info was added; error otherwise.
-func (is *infoStore) addInfo(key string, i *info) error {
+func (is *infoStore) addInfo(key string, i *Info) error {
 	// Only replace an existing info if new timestamp is greater, or if
 	// timestamps are equal, but new hops is smaller.
 	var contentsChanged bool
 	if existingInfo, ok := is.Infos[key]; ok {
-		if i.Timestamp < existingInfo.Timestamp ||
-			(i.Timestamp == existingInfo.Timestamp && i.Hops >= existingInfo.Hops) {
+		var (
```

Comment:
Strange use of a vars section...inside a block like this, I'm more accustomed to seeing:

			iNanos := i.Value.Timestamp.WallTime
			existingNanos := existingInfo.Value.Timestamp.WallTime